lib/data/integrity_checker.rb

```diff
@@ -0,0 +1,29 @@
+require 'auth/authentication_backend'
+# The purpose of this is to make sure any existing Cognito groups
+# have a corresponding local team in DB. If not, it creates them.
+
+class IntegrityChecker
+  include AuthenticationBackend
+
+  def initialize
+    return if SelfService.service(:cognito_stub)
+
+    Rails.logger.info('Checking the data integrity...')
+    check_groups_vs_teams
+  end
+
+  def check_groups_vs_teams
+    cognito_groups = list_groups
+    return if cognito_groups.empty?
+
+    cognito_groups.groups.each do |group|
+      Rails.logger.info("Checking if #{group.group_name} group has a corresponding team in DB...")
+      if Team.exists?(team_alias: group.group_name)
+        Rails.logger.info("OK - #{group.group_name} group has a corresponding team in DB...")
+      else
+        Rails.logger.warn("#{group.group_name} group does not have a corresponding team in DB! Creating...")
+        NewTeamEvent.create(name: group.description)
```

Comment:
Do groups always have a description when created in cognito?

Reply:
yes, https://github.com/alphagov/verify-self-service/blob/c259f61d57c1aa000783138683c58ba00ee93047/app/models/new_team_event.rb#L28

Reply:
good point though, if someone created them manually in the AWS Console and didn't add it.. which they shouldn't do! ... I'll put a check for it then

Reply:
added the check + test in https://github.com/alphagov/verify-self-service/pull/220/commits/9ef2bfb65bce930c09c08ed96d81d67902a4890f

Reply:
ah yeah I meant in the console :) Awesome! 👍